2022-12-27 14:28:28.561 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2022-12-27 14:28:34.434 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2022-12-27 14:29:14.039 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1866), thread 'MainThread' (140707879885888):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2022-12-28 09:47:48.807 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1817), thread 'MainThread' (139950225042304):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 81, in get_label
    self.prompt = frame_challenge.locator("//h2[@class='prompt-text']").text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 15059, in text_content
    self._sync(self._impl_obj.text_content(timeout=timeout))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 526, in text_content
    return await self._frame.text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 598, in text_content
    return await self._channel.send("textContent", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 10000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//iframe[contains(@title, 'hCaptcha挑战')]"
  selector resolved to hidden <iframe scrolling="no" frameborder="0" aria-hidden="true…></iframe>
waiting for selector "//iframe[contains(@title, 'hCaptcha挑战')] >> internal:control=enter-frame >> //h2[@class='prompt-text']"
============================================================

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 251, in anti_hcaptcha
    self.get_label(frame_challenge)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 85, in get_label
    raise ChallengePassed("Man-machine challenge unexpectedly passed")
hcaptcha_challenger.exceptions.ChallengePassed: Message: Man-machine challenge unexpectedly passed

2022-12-29 09:52:25.462 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2022-12-29 09:52:31.344 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2022-12-29 09:53:11.681 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1911), thread 'MainThread' (140628557065280):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2022-12-30 09:52:30.692 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2022-12-30 09:52:36.233 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2022-12-30 09:53:22.590 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1968), thread 'MainThread' (140149442538560):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2022-12-31 09:49:34.429 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2022-12-31 09:49:40.370 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2022-12-31 09:50:20.615 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1907), thread 'MainThread' (140559780953152):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-01 10:06:45.032 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-01 10:06:52.880 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-01 10:07:34.734 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1921), thread 'MainThread' (139980530441088):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-02 09:51:10.520 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-02 09:51:16.502 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-02 09:51:57.089 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1890), thread 'MainThread' (139818514111552):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-03 09:47:04.171 | ERROR    | hcaptcha_challenger._solutions.kernel:request_assets:157 - 0
2023-01-03 09:47:22.824 | ERROR    | hcaptcha_challenger._solutions.kernel:request_assets:157 - 0
2023-01-03 09:47:23.049 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1872), thread 'MainThread' (139971826125888):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 267, in anti_hcaptcha
    self.challenge(frame_challenge, model=model)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 125, in challenge
    result = model.solution(img_stream=data, label=self.label_alias[self.label])
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/hcaptcha_challenger/_solutions/resnet.py", line 63, in solution
    return self.classifier(img_stream, feature_filters=None)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/hcaptcha_challenger/_solutions/resnet.py", line 54, in classifier
    raise ResourceWarning(_err_prompt)  # maybe mercy
ResourceWarning: 
            The remote network does not exist or the local cache has expired.
            1. Check objects.yaml for typos | model=rose.onnx;
            2. Restart the program after deleting the local cache | dir=datas/models/_assets;
            
2023-01-04 09:53:43.820 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-04 09:53:49.927 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-04 09:54:30.593 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1913), thread 'MainThread' (139851197082688):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-05 09:56:43.082 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-05 09:56:51.661 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-05 09:57:34.606 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1816), thread 'MainThread' (140110985169792):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-06 09:57:50.905 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-06 09:57:56.432 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-06 09:58:37.907 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1955), thread 'MainThread' (140568463334464):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-07 09:49:38.707 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1951), thread 'MainThread' (140291671489408):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 81, in get_label
    self.prompt = frame_challenge.locator("//h2[@class='prompt-text']").text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 15059, in text_content
    self._sync(self._impl_obj.text_content(timeout=timeout))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 526, in text_content
    return await self._frame.text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 598, in text_content
    return await self._channel.send("textContent", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 10000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//iframe[contains(@title, 'hCaptcha挑战')]"
  selector resolved to hidden <iframe scrolling="no" frameborder="0" aria-hidden="true…></iframe>
waiting for selector "//iframe[contains(@title, 'hCaptcha挑战')] >> internal:control=enter-frame >> //h2[@class='prompt-text']"
============================================================

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 251, in anti_hcaptcha
    self.get_label(frame_challenge)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 85, in get_label
    raise ChallengePassed("Man-machine challenge unexpectedly passed")
hcaptcha_challenger.exceptions.ChallengePassed: Message: Man-machine challenge unexpectedly passed

2023-01-08 10:00:02.594 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-08 10:00:08.994 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-08 10:00:50.879 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1943), thread 'MainThread' (139773863328832):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-09 09:54:35.979 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-09 09:54:41.361 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-09 09:55:22.299 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1808), thread 'MainThread' (140232298662976):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-10 09:55:43.637 | ERROR    | hcaptcha_challenger._solutions.kernel:request_assets:157 - 0
2023-01-10 09:56:14.586 | ERROR    | hcaptcha_challenger._solutions.kernel:request_assets:157 - 0
2023-01-10 09:56:14.734 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1955), thread 'MainThread' (140667217185856):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 267, in anti_hcaptcha
    self.challenge(frame_challenge, model=model)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 125, in challenge
    result = model.solution(img_stream=data, label=self.label_alias[self.label])
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/hcaptcha_challenger/_solutions/resnet.py", line 63, in solution
    return self.classifier(img_stream, feature_filters=None)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/hcaptcha_challenger/_solutions/resnet.py", line 54, in classifier
    raise ResourceWarning(_err_prompt)  # maybe mercy
ResourceWarning: 
            The remote network does not exist or the local cache has expired.
            1. Check objects.yaml for typos | model=ladybug.onnx;
            2. Restart the program after deleting the local cache | dir=datas/models/_assets;
            
2023-01-11 09:56:23.810 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-11 09:56:30.321 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-11 09:57:12.280 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1881), thread 'MainThread' (140521175563328):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-12 09:55:49.061 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-12 09:55:54.261 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-12 09:56:35.340 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1817), thread 'MainThread' (140488860232768):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-13 10:00:03.180 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-13 10:00:09.415 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-13 10:00:52.094 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1803), thread 'MainThread' (139647707585600):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-14 09:50:17.994 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-14 09:50:22.854 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-14 09:51:02.922 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1785), thread 'MainThread' (139630694857792):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-15 09:59:04.207 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-15 09:59:09.331 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-15 09:59:49.854 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1851), thread 'MainThread' (139730961140800):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-16 09:55:19.014 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-16 09:55:24.158 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-16 09:56:05.043 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1924), thread 'MainThread' (140188109315136):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-17 09:51:31.660 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1850), thread 'MainThread' (140573534493760):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 81, in get_label
    self.prompt = frame_challenge.locator("//h2[@class='prompt-text']").text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 15059, in text_content
    self._sync(self._impl_obj.text_content(timeout=timeout))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 526, in text_content
    return await self._frame.text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 598, in text_content
    return await self._channel.send("textContent", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 10000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//iframe[contains(@title, 'hCaptcha挑战')]"
  selector resolved to hidden <iframe scrolling="no" frameborder="0" aria-hidden="true…></iframe>
waiting for selector "//iframe[contains(@title, 'hCaptcha挑战')] >> internal:control=enter-frame >> //h2[@class='prompt-text']"
============================================================

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 251, in anti_hcaptcha
    self.get_label(frame_challenge)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 85, in get_label
    raise ChallengePassed("Man-machine challenge unexpectedly passed")
hcaptcha_challenger.exceptions.ChallengePassed: Message: Man-machine challenge unexpectedly passed

2023-01-18 09:59:28.090 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-18 09:59:34.234 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-18 10:00:15.782 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1856), thread 'MainThread' (140553062931328):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-20 09:58:26.862 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-20 09:58:33.699 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-20 09:59:15.246 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1786), thread 'MainThread' (140718030937152):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-21 09:53:27.588 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-21 09:53:32.809 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-21 09:54:13.421 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1967), thread 'MainThread' (140127924005952):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-22 09:56:07.279 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1952), thread 'MainThread' (139637560609856):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 81, in get_label
    self.prompt = frame_challenge.locator("//h2[@class='prompt-text']").text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 15059, in text_content
    self._sync(self._impl_obj.text_content(timeout=timeout))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 526, in text_content
    return await self._frame.text_content(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 598, in text_content
    return await self._channel.send("textContent", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 10000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//iframe[contains(@title, 'hCaptcha挑战')]"
  selector resolved to hidden <iframe scrolling="no" frameborder="0" aria-hidden="true…></iframe>
waiting for selector "//iframe[contains(@title, 'hCaptcha挑战')] >> internal:control=enter-frame >> //h2[@class='prompt-text']"
============================================================

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 251, in anti_hcaptcha
    self.get_label(frame_challenge)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 85, in get_label
    raise ChallengePassed("Man-machine challenge unexpectedly passed")
hcaptcha_challenger.exceptions.ChallengePassed: Message: Man-machine challenge unexpectedly passed

2023-01-23 09:50:43.543 | ERROR    | hcaptcha_challenger._solutions.kernel:request_assets:157 - 0
2023-01-23 09:51:03.431 | ERROR    | hcaptcha_challenger._solutions.kernel:request_assets:157 - 0
2023-01-23 09:51:03.559 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1919), thread 'MainThread' (140285211262016):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 708, in refresh_ctx_cookies
    resp = self.armor.anti_hcaptcha(page, window="login", recur_url=recur_url)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 267, in anti_hcaptcha
    self.challenge(frame_challenge, model=model)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 125, in challenge
    result = model.solution(img_stream=data, label=self.label_alias[self.label])
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/hcaptcha_challenger/_solutions/resnet.py", line 63, in solution
    return self.classifier(img_stream, feature_filters=None)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/hcaptcha_challenger/_solutions/resnet.py", line 54, in classifier
    raise ResourceWarning(_err_prompt)  # maybe mercy
ResourceWarning: 
            The remote network does not exist or the local cache has expired.
            1. Check objects.yaml for typos | model=sunflower.onnx;
            2. Restart the program after deleting the local cache | dir=datas/models/_assets;
            
2023-01-24 09:55:38.159 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-24 09:55:45.235 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-24 09:56:27.181 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1927), thread 'MainThread' (140032775580544):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-25 09:53:32.958 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-25 09:53:38.558 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-25 09:54:18.957 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1930), thread 'MainThread' (139796986747968):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
2023-01-26 09:55:33.590 | CRITICAL | services.bricklayer.core:refresh_ctx_cookies:717 - >> MISS [CookieManager] Identity token update failed
2023-01-26 09:55:40.040 | ERROR    | services.explorer.explorer:get_order_history:142 - Failed to get order history, cookie may have expired
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 301, in run
    promotions = self.preload()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 269, in preload
    self.ph.get_oder_history(ctx_cookies=self._ctx_cookies)
  File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 219, in get_oder_history
    if resp := explorer.get_order_history(ctx_cookies):
> File "/home/runner/work/gamer/gamer/epic/src/services/explorer/explorer.py", line 140, in get_order_history
    raise RequestException("Failed to get order history, cookie may have expired")
requests.exceptions.RequestException: Failed to get order history, cookie may have expired
2023-01-26 09:56:21.560 | ERROR    | services.scaffold:claim:138 - An error has been caught in function 'claim', process 'MainProcess' (1865), thread 'MainThread' (140381696298048):
Traceback (most recent call last):
  File "/home/runner/work/gamer/gamer/epic/src/main.py", line 11, in <module>
    Fire(Scaffold)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 466, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
> File "/home/runner/work/gamer/gamer/epic/src/services/scaffold.py", line 138, in claim
    claimer.run(silence=silence, log_ignore=ignore, unreal=unreal)
  File "/home/runner/work/gamer/gamer/epic/src/apis/scaffold/claimer.py", line 26, in run
    ClaimerScheduler(silence=silence, unreal=unreal).job_loop_claim(log_ignore)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 112, in job_loop_claim
    claimer.just_do_it()
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 309, in just_do_it
    fire(
  File "/home/runner/work/gamer/gamer/epic/src/services/utils/toolbox.py", line 84, in fire
    container(context)
  File "/home/runner/work/gamer/gamer/epic/src/services/deploy.py", line 305, in run
    result = empower_games_claimer(self.bricklayer, promotion.url, page, pattern="get")
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 170, in empower_games_claimer
    return claimer.get_free_game(page_link=page_link, page=page, pattern=pattern)
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/game.py", line 157, in get_free_game
    self.cart_handle_payment(page)  # mode: get
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 557, in cart_handle_payment
    if self._click_order_button(page):  # cart_handle_payment
  File "/home/runner/work/gamer/gamer/epic/src/services/bricklayer/core.py", line 456, in _click_order_button
    payment_btn.click()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13519, in click
    self._sync(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_locator.py", line 149, in click
    return await self._frame.click(self._selector, strict=True, **params)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py", line 484, in click
    await self._channel.send("click", locals_to_params(locals()))
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 396, in wrap_api_call
    return await cb()
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
  waiting for frame "xpath=//div[@id='webPurchaseContainer']//iframe"
============================================================
